src/editors/containers/VideoEditor/components/VideoSettingsModal/components/VideoSourceWidget/hooks.jsx

```diff
@@ -1,70 +1,27 @@
 import { actions } from '../../../../../../data/redux';
-import { isEdxVideo } from '../../../../../../data/services/cms/api';
 
-/**
- * updateVideoId({ dispatch })({e, source})
- * updateVideoId takes the current onBlur event, the current object of the video
- * source, and dispatch  method, and updates the redux value for all the fields to
- * their default values except videoId, fallbackVideos, and handouts.
- * @param {event} e - object for onBlur event
- * @param {func} dispatch - redux dispatch method
- * @param {object} source - object for the Video Source field functions and values
- */
-export const updateVideoId = ({ dispatch }) => ({ e, source }) => {
-  if (source.local !== '') {
-    if (source.formValue !== e.target.value) {
-      source.onBlur(e);
-      let videoId;
-      let videoSource;
-      if (isEdxVideo(source.local)) {
-        videoId = source.local;
-        videoSource = '';
-      } else if (source.local.includes('youtu.be') || source.local.includes('youtube')) {
-        videoId = '';
-        videoSource = source.local;
+export const sourceHooks = ({ dispatch }) => ({
+  updateVideoURL: ({ e }) => dispatch(actions.video.updateField({ videoSource: e.target.value, videoId: '' })),
+  updateVideoId: ({ e }) => dispatch(actions.video.updateField({ videoId: e.target.value, videoSource: '' })),
+});
+
+export const fallbackHooks = ({ fallbackVideos, dispatch }) => ({
+  addFallbackVideo: () => dispatch(actions.video.updateField({ fallbackVideos: [...fallbackVideos, ''] })),
+  deleteFallbackVideo: (videoUrl) => {
+    const updatedFallbackVideos = [];
+    let firstOccurence = true;
+    fallbackVideos.forEach(item => {
```

Comment:
```suggestion
    const updatedFallbackVideos = fallbackVideos.splice(fallbackVideos.indexOf(videoUrl),1)
```
Does the same thing, is cleaner.